lib/galaxy/web/stack/__init__.py

```diff
@@ -14,59 +17,243 @@
 except ImportError:
     uwsgi = None
 
-try:
-    from uwsgidecorators import postfork as uwsgi_postfork
-except:
-    uwsgi_postfork = lambda x: x    # noqa: E731
-    if uwsgi is not None and hasattr(uwsgi, 'numproc'):
-        print("WARNING: This is a uwsgi process but the uwsgidecorators library"
-              " is unavailable.  This is likely due to using an external (not"
-              " in Galaxy's virtualenv) uwsgi and you may experience errors. "
-              "HINT:\n  {venv}/bin/pip install uwsgidecorators".format(
-                  venv=os.environ.get('VIRTUAL_ENV', '/path/to/venv')))
+from six import string_types
+
+from .message import ApplicationStackMessage, ApplicationStackMessageDispatcher, JobHandlerMessage, decode
+from .transport import ApplicationStackTransport, UWSGIFarmMessageTransport
+
+from galaxy.util.bunch import Bunch
 
 
 log = logging.getLogger(__name__)
 
 
+class ApplicationStackLogFilter(logging.Filter):
+    def filter(self, record):
+        record.worker_id = None
+        record.mule_id = None
+        return True
+
+
+class UWSGILogFilter(logging.Filter):
+    def filter(self, record):
+        record.worker_id = uwsgi.worker_id()
+        record.mule_id = uwsgi.mule_id()
+        return True
+
+
 class ApplicationStack(object):
     name = None
     prohibited_middleware = frozenset()
+    transport_class = ApplicationStackTransport
+    log_filter_class = ApplicationStackLogFilter
+    # TODO: this belongs in the pool configuration
+    server_name_template = '{server_name}'
+
+    # used both to route jobs to a pool with this name and indicate whether or
+    # not a stack is using messaging for handler assignment
+    pools = Bunch(
+        JOB_HANDLERS = 'job-handlers',
+    )
 
     @classmethod
     def register_postfork_function(cls, f, *args, **kwargs):
         f(*args, **kwargs)
 
-    def workers(self):
-        return []
+    def __init__(self, app=None):
+        self.app = app
+
+    def start(self):
+        # TODO: with a stack config the pools could be parsed here
+        pass
 
     def allowed_middleware(self, middleware):
         if hasattr(middleware, '__name__'):
             middleware = middleware.__name__
         return middleware not in self.prohibited_middleware
 
+    def workers(self):
+        return []
+
+    @property
+    def pool_name(self):
+        # TODO: ideally jobs would be mappable to handlers by pool name
+        return None
+
+    def has_pool(self, pool_name):
+        return False
+
+    def in_pool(self, pool_name):
+        return False
+
+    def set_server_name(self, app, caller_tmpl_dict):
+        tmpl_dict = {
+            'server_name': app.config.server_name,
+            'pool_name': self.pool_name,
+            'process_num': 1,
+            'id': 1,
+            'hostname': socket.gethostname().split('.', 1)[0],
+            'fqdn': socket.getfqdn(),
```

Comment:
Ah, you meant the FQDN specifically. I think I'd rather leave `fqdn` as it is and add an `infrastructure_hostname` that would be parsed out of the infrastructure URL. FQDN is going to be more useful in cases like mine where you are running one Galaxy instance across multiple hosts. The infrastructure hostname is going to be the same on all of those.